lib/vrf.c

```diff
@@ -326,6 +326,34 @@ void vrf_disable(struct vrf *vrf)
 		(*vrf_master.vrf_disable_hook)(vrf);
 }
 
+/* Disable VRF module. */
+void vrf_disable_all(void)
```

Comment:
hmm - so this sort of duplicates the logic in `vrf_terminate` ? it would be cleaner to have a single path, imo, that "knows" how to iterate through all the vrfs. That single path could have a 'terminate' boolean, to distinguish between the "disable-only" and "terminate" cases?

Reply:
vrf_terminate() is called by a number of subsystems, bgp is an example. So I didn't want to change the signature of vrf_terminate() to affect multiple calls

Reply:
quite - I wasn't suggesting changing the existing signature, but something like:
- add an internal function that can do the iteration and disable and/or terminate
- add new api that uses that common function to 'disable' only
- convert 'terminate' to use the common function with 'terminate'

Reply:
No other subsystem, except zebra, needs a separate vrf_disable() part alone, and initially I was looking into making vrf_disable_all() call just a zebra specific function zebra_vrf_disable_all(), and place it in zebra/zebra_vrf.c. 
I would rather do that if I asked.

Reply:
I think we should avoid a second copy of the iteration logic that knows how to iterate, knows about the special case for the default vrf, etc. I think there should only be one copy of that iteration code, if at all possible - and I think it is possible?
I would certainly not want one copy of the code in the library, and then a cut-and-paste in zebra.